addOns/grpc/src/main/java/org/zaproxy/addon/grpc/internal/EncoderUtils.java

```diff
@@ -0,0 +1,302 @@
+        if (str.startsWith("\"") && str.endsWith("\"")) {
+            // Remove the first and last characters
+            str = str.substring(1, str.length() - 1);
+        } else {
+            throw new InvalidProtobufFormatException(
+                    "Invalid Format: String must be enclosed by double quotes.");
+        }
+        return str;
+    }
+
+    static String[] validateAndSplitInput(String input) throws InvalidProtobufFormatException {
+        String[] inputArray = input.split("::", 2);
+        if (inputArray.length == 0) {
+            throw new InvalidProtobufFormatException(
+                    "Invalid Format: Missing field number and Wire type");
+        }
+        String[] fieldNumWireType = inputArray[0].split(":", 2);
+        if (fieldNumWireType.length != 2) {
+            throw new InvalidProtobufFormatException(
+                    "Invalid Format: Missing field number or Wire type");
+        }
+        return fieldNumWireType;
+    }
+
+    static int getWiretype(String tag) {
+        if (tag.length() > 1) {
+            return tag.charAt(0) - '0';
+        } else {
+            return Integer.parseInt(tag);
+        }
+    }
+
+    static char getTypeSpecifier(String tag) {
+        if (tag.length() > 1) {
+            return tag.charAt(1);
+        } else {
+            // default value, give no information
+            return 'n';
+        }
+    }
+
+    static void writeFields(List<String> inputString, CodedOutputStream codedOutputStream)
+            throws IOException, InvalidProtobufFormatException {
+        for (String input : inputString) {
+            String[] inputArray = input.split("::", 2);
+
+            String[] fieldNumWireType = validateAndSplitInput(input);
+
+            String tag = fieldNumWireType[1];
+            int wireType = getWiretype(tag);
+            char typeSpecifier = getTypeSpecifier(tag);
+
+            int fieldNumber = Integer.parseInt(fieldNumWireType[0]);
+
+            switch (wireType) {
+                case VARINT_WIRE_TYPE:
+                    long value = Long.parseLong(inputArray[1]);
+                    codedOutputStream.writeInt64(fieldNumber, value);
+                    break;
+                case BIT64_WIRE_TYPE:
+                    if (typeSpecifier == 'D') {
+                        codedOutputStream.writeDouble(
+                                fieldNumber, Double.parseDouble(inputArray[1]));
+                    } else {
+                        codedOutputStream.writeFixed64(fieldNumber, Long.parseLong(inputArray[1]));
+                    }
+                    break;
+                case LENGTH_DELIMITED_WIRE_TYPE:
+                    String val = inputArray[1];
+
+                    if (typeSpecifier == 'B') {
+                        byte[] byteArray = hexStringtoByteArray(inputArray[1]);
+                        codedOutputStream.writeByteArray(fieldNumber, byteArray);
+                    } else if (typeSpecifier == 'N') {
```

Comment:
Are these supposed to be case sensitive? What if it's a lower n like back on line 151?

Reply:
Currently lowercase n means nothing, no information about wiretype, yes it is case sensitive, I can make this case-insensitive, then I will change n to something else

Reply:
🤷‍♂️ you know the spec better than I 😃